src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -409,7 +421,7 @@ public boolean isLegacyWatchApp() {
     }
 
     if (!frameworks.isEmpty()) {
-      Path frameworksDestinationPath = bundleRoot.resolve(this.destinations.getFrameworksPath());
```

Comment:
I mean, this whole block https://github.com/AquaGeek/buck/blob/SwiftSupport/src/com/facebook/buck/apple/BuiltinApplePackage.java#L123-L147 could be moved in AppleBundle, then your comment about grabbing `swiftStdlibTool` from the `bundle` instance will also be resolved, isn't it?